e2e/registration_test.go

```diff
@@ -200,3 +201,32 @@ func getServiceInstancesByName(t testing.TB, srv *testutil.TestServer, serviceNa
 	filter := fmt.Sprintf(`Service == "%s"`, serviceName)
 	return testutils.ListServices(t, srv, filter)
 }
+
+func waitForServiceByName(t testing.TB, srv *testutil.TestServer, serviceName string) (map[string]testutils.Service, error) {
+	var instances map[string]testutils.Service
+	polling := make(chan struct{})
+	stopPolling := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-stopPolling:
+				return
```

Comment:
maybe add a tiny sleep to free up resources

Reply:
Sure thing! I've got some other polls, I will add them there too.